raiden/network/transport/matrix/transport.py

```diff
@@ -1012,12 +1015,48 @@ def _get_room_for_address(
             self.log.error('No valid peer found', peer_address=address_hex)
             return None
 
-        self._address_to_userids[address].update({user.user_id for user in peers})
-
         if self._private_rooms:
             room = self._get_private_room(invitees=peers)
         else:
             room = self._get_public_room(room_name, invitees=peers)
+
+        peer_ids = self._address_to_userids[address]
+        member_ids = {member.user_id for member in room.get_joined_members()}
+        room_is_empty = not bool(peer_ids & member_ids)
+        if room_is_empty:
+            last_ex: Optional[Exception] = False
+            retry_interval = 0.1
+            self.log.debug(
+                'Waiting for peer to join from invite.',
+                peer_address=address_hex,
+            )
+            for _ in range(JOIN_RETRIES):
+                try:
+                    member_ids = {member.user_id for member in room.get_joined_members()}
```

Comment:
IIRC, there's a ttlcache on `room.get_joined_members` which would not fetch the results again in 10secs, which may defeat the purpose of this loop. You could check on how to rely on the events propagation/processing instead, or remove the cache to bash the server for room membership updates

Reply:
Ah thx, so the chache is why the tests that require fast server responses are failing now. Any suggestion if/which events get propagated, if a user joins a room? Otherwise I'll just go for clearing the cache

Reply:
Removed the caching and added a explicit flag instead.